benchmarking/prototype_parameterized_template.py

```
@@ -0,0 +1,613 @@
+        }
+        call_values = []
+        for variable in template_inputs:
+            constant_index = placeholder_names.get(variable.name)
+            if constant_index is not None:
+                call_values.append(np.asarray(item["constants"][constant_index].data))
+            else:
+                role = _normalize(variable.name)
+                if role not in values_by_role:
+                    raise KeyError(
+                        f"No runtime input role {role!r} for analysis {item['name']}"
+                    )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)